src/routes/return.js

```diff
@@ -1,52 +1,52 @@
+import { ethers } from 'ethers';
 import { useState, useEffect, useRef, useCallback } from 'react';
 import { Container } from 'react-bootstrap';
 import NFTCardGrid from '../components/NFTCardGrid';
 import LoginService from '../utils/LoginService';
+import {KasuContract, ERC721_ABI} from '../utils/abiManager';
 
 function ReturnPage() {
     const [rentedNFTs, setRentedNFTs] = useState([]);
-    const fetchRentedNFTs = useCallback(() => {
+    const returnNFT = async function(listing){
+        const loginService = LoginService.getInstance();
+        const tokenContract = new ethers.Contract(listing.tokenAddress, ERC721_ABI(), loginService.signer);
+        await tokenContract.approve(listing.lenderAddress, listing.tokenId)
```

Comment:
During local debugging this causes an error like:
```
eth_call
  Contract call:       <UnrecognizedContract>
  From:                0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266
  To:                  0xf953b3a269d80e3eb0f2947630da976b896a8c5b

  Error: Transaction reverted without a reason string
      at <UnrecognizedContract>.<unknown> (0xf953b3a269d80e3eb0f2947630da976b896a8c5b)
      at HardhatNode.runCall (/Users/amar/dev/personal/crypto/nft-rental-marketplace/node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:535:20)
      at EthModule._callAction (/Users/amar/dev/personal/crypto/nft-rental-marketplace/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:353:9)
      at HardhatNetworkProvider._sendWithLogging (/Users/amar/dev/personal/crypto/nft-rental-marketplace/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:138:22)
      at HardhatNetworkProvider.request (/Users/amar/dev/personal/crypto/nft-rental-marketplace/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:115:18)
      at JsonRpcHandler._handleRequest (/Users/amar/dev/personal/crypto/nft-rental-marketplace/node_modules/hardhat/src/internal/hardhat-network/jsonrpc/handler.ts:188:20)
      at JsonRpcHandler._handleSingleRequest (/Users/amar/dev/personal/crypto/nft-rental-marketplace/node_modules/hardhat/src/internal/hardhat-network/jsonrpc/handler.ts:167:17)
      at Server.JsonRpcHandler.handleHttp (/Users/amar/dev/personal/crypto/nft-rental-marketplace/node_modules/hardhat/src/internal/hardhat-network/jsonrpc/handler.ts:52:21)

eth_sendRawTransaction
  Contract call:       <UnrecognizedContract>
  Transaction:         0x590067620dfd03c85037ebd1fcb890e6a90c165828c3a6d5f7713f90167e78e0
  From:                0x976ea74026e726554db657fa54763abd0c3a0aa9
  To:                  0xf953b3a269d80e3eb0f2947630da976b896a8c5b
  Value:               0 ETH
  Gas used:            29345 of 29345
  Block #176:          0x9f90ce13745b5dc573a768ebe23c5faf5a3ee518f0239dd3feec223fc894429f
```
Right as metamask pops up for the approval, and before anything is selected by the user.

Reply:
I tried your branch and there was no problem approving

![image](https://user-images.githubusercontent.com/4744406/151189546-7d9ef3cb-c46f-4392-be08-6eacb5c00b57.png)

Try to deploy again? I think your local contract address might be incorrect.

Reply:
Note that you should approve Kasu contract's the address, not the lender's address.